src/zenml/post_execution/pipeline_run.py

```diff
@@ -169,11 +177,34 @@ def get_step(self, name: str) -> StepView:
             KeyError: If there is no step with the given name.
         """
         self._ensure_steps_fetched()
+        if step and step_name:
+            raise TypeError(
+                "'step' and 'step_name' both set for the"
+                "get_step() method. This is not supported. "
+                "Please set either the 'step' or the "
+                "'step_name' parameter."
+            )
+        elif step:
+            if isinstance(step, zenml.steps.base_step.BaseStepMeta):
```

Comment:
I'm not sure why this check is needed here? Is it to also allow step classes being passed?

If yes I suggest we replace it by `issubclass(step, BaseStep` instead of a check of the metaclass. It would also need to be included in the type annotation of the function signature, e.g. 
```python
step: Optional[Union["BaseStep", Type["BaseStep"]] = None,
```

I also have one more "suggestion" (I'm not quite sure if I like it myself), but just throwing it our there in case you like it:
Instead of having two parameters `step_name` and `step`, we could make it one parameter which can be either a string or a step. I like two things about it:
- We don't need a check if both/none of them are set
- Usage of the method is much easier and doesn't require a keyword argument each time:
```python
# both of these work without any step=..., step_name=...
run.get_step(MyStepClass)
run.get_step("name_of_my_step")
```